packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -668,11 +669,17 @@ class DirectStripeRoutes {
     }
 
     const { priceId, paymentMethodId, idempotencyKey } = request.payload;
+    const PMI = paymentMethodId || extractPaymentMethodId(customer);
+
+    if (!PMI) {
+      throw error.paymentFailed();
+    }
+
     const subIdempotencyKey = `${idempotencyKey}-createSub`;
     const subscription = await this.stripeHelper.createSubscriptionWithPMI(
       customer.id,
       priceId,
-      paymentMethodId,
+      PMI,
```

Comment:
This works? It looks like it can end up attempting to attach a source as a payment method. Should we just skip attaching the payment method in the `createSubscriptionWithPMI` call if we already have a source or payment method on file?

Reply:
It works.  But you are probably right about the attaching the source as a payment method step being unnecessary.  I'll verify and make changes.

Reply:
I was thinking the payment method could be an optional arg. The easiest way to accommodate that would be turning the args into an object and making the payment method optional in TypeScript. We should really consider having a policy that as soon as a function passes 2 args, we should switch to an object. All the libraries typically do this, and it'd make it easier to avoid messing up positional placement of args.